File: Interpreter/src/model/ArithExpresion.java
package model;

import DataStructure.IntHeap;
import DataStructure.MyIntDict;
import ExceptionHandling.MyException;



public class ArithExpresion extends Expresion {
    Expresion first;
    Expresion second;
    int op;

    public ArithExpresion(int _op,
                          Expresion _first,
                          Expresion _second){
        first = _first;
        second = _second;
        op = _op;

    }

    @Override
    int eval(MyIntDict<String, Integer> tab, IntHeap heap) throws MyException{
        int val = 0;

        switch (op){
            case 1:{
                val = first.eval(tab, heap) + second.eval(tab, heap);
                break;
            }
            case 2:{
                val = first.eval(tab, heap) - second.eval(tab, heap);
                break;
            }
            case 3:{
                val = first.eval(tab, heap) * second.eval(tab, heap);
                break;
            }
            case 4:{
                int divisor = second.eval(tab, heap);
                if(divisor == 0){
                    throw new MyException("Divide by zero");
                }
                val = first.eval(tab, heap) / divisor;
                break;
            }
            default:
                break;

        }
        return val;
    }

    @Override
    public String toString() {
        String operator = "";
        switch (op) {
            case 1: {
                operator = "+";
                break;
            }
            case 2: {
                operator = "-";
                break;
            }
            case 3: {
                operator = "*";
                break;
            }
            case 4: {
                operator = "/";
                break;
            }
            default:
                break;
        }
        return  first.toString() + operator + second.toString();
    }
}

File: map/Interpreter2.0/model/ArithExpresion.cs
using DataStructure;
using MyException;
using System;
namespace models
{
    public class ArithExpresion : Expression
    {
        Expression first;
        Expression second;
        int op;

        public ArithExpresion(int _op,
                            Expression _first,
                            Expression _second){
            first = _first;
            second = _second;
            op = _op;

        }

        override public int eval(MyIntDict<string, int> tab){
            int val = 0;

            switch (op){
                case 1:{
                    val = first.eval(tab) + second.eval(tab);
                    break;
                }
                case 2:{
                    val = first.eval(tab) - second.eval(tab);
                    break;
                }
                case 3:{
                    val = first.eval(tab) * second.eval(tab);
                    break;
                }
                case 4:{
                    int divisor = second.eval(tab);
                    if(divisor == 0){
                        throw new MyException.MyException("Divide by zero");
                    }
                    val = first.eval(tab) / divisor;
                    break;
                }
                default:
                    break;

            }
            return val;
        }

       
        override public String ToString() {
            String oper = "";
            switch (op) {
                case 1: {
                    oper = "+";
                    break;
                }
                case 2: {
                    oper = "-";
                    break;
                }
                case 3: {
                    oper = "*";
                    break;
                }
                case 4: {
                    oper = "/";
                    break;
                }
                default:
                    break;
            }
            return  first.ToString() + oper + second.ToString();
        }
    }
}